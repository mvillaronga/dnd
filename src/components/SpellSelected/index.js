import React from 'react'

const SpellSelected = ({selection, onSpellDeselected}) => {
  return (
    <React.Fragment>
      {selection.map((spell, idx) => 
        <span onClick={() => onSpellDeselected(spell)}>
          {spell.name}{(idx < selection.length - 1) ? ', ' : ''}
        </span>)
      }
      {selection.length > 0 && <hr />}
    </React.Fragment>
  )
}

export default SpellSelected