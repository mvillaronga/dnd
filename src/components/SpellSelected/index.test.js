import React from 'react';
//import SpellList from '../SpellList';
//import SpellCard from '../SpellCard'
import SpellSelected from '../SpellSelected'
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

test('Default Spell Card Rendering', () => {
    const component = renderer.create(
      <SpellSelected 
        selection={[{name:"Test Spell 1", level:"1"}, {name:"Test Spell 2", level:"9"}]} 
        onSpellDeselected={ jest.fn() } 
        onClearSelection={ jest.fn() } />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Its children should handle a click', () => {
  const click = jest.fn()
  const output = shallow(
    <SpellSelected 
    selection={[{name:"Test Spell 1", level:"1"}, {name:"Test Spell 2", level:"9"}]} 
    onSpellDeselected={ click } 
    onClearSelection={ jest.fn() } />,
  )
  output.find("span").first().simulate('click')
  expect(click).toHaveBeenCalled()
  })

  test('It handle clicking the clear button', () => {
    const click = jest.fn()
    const output = shallow(
      <SpellSelected 
      selection={[{name:"Test Spell 1", level:"1"}, {name:"Test Spell 2", level:"9"}]} 
      onSpellDeselected={ jest.fn() } 
      onClearSelection={ click } />,
    )
    output.find("button").first().simulate('click')
    expect(click).toHaveBeenCalled()
    })