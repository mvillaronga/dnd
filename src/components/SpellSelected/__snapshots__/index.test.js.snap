// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Spell Card Rendering 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Clear Selection
  </button>
  <div>
    <span
      onClick={[Function]}
    >
      Test Spell 1
      , 
    </span>
    <span
      onClick={[Function]}
    >
      Test Spell 2
      
    </span>
  </div>
  <hr />
</div>
`;
